libs/journeys/ui/src/components/SearchBar/SearchDropdown/CountryLanguageSelector/CountryLanguageSelector.tsx

```diff
@@ -52,31 +52,25 @@ export function CountryLanguageSelector({
     }
   }
 
-  function getCountryCookie(): string | undefined {
-    return document.cookie
-      .split('; ')
-      .find((row) => row.startsWith('NEXT_COUNTRY'))
-      ?.split('=')[1]
-  }
-
   function parseCountryName(countryCode: string): string | undefined {
     const displayNames = new Intl.DisplayNames(['en'], { type: 'region' })
     return displayNames.of(countryCode)
   }
 
-  const findUserCountry = useCallback(() => {
-    const countryCookie = getCountryCookie()
-    if (countryCookie != null) {
-      const [, countryCode] = countryCookie.split('---')
-      const countryName = parseCountryName(countryCode)
-      setCountry(countryName)
-      setCountryCode(countryCode)
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  const findUserCountry = useCallback(async () => {
+    await fetch('/api/jf/watch.html/geolocation').then(async (response) => {
+      await response.json().then((data: { country?: string }) => {
+        if (data.country != null) {
+          const countryName = parseCountryName(data.country)
+          setCountry(countryName)
+          setCountryCode(data.country)
+        }
+      })
+    })
```

Comment:
Was thinking more can it be done like...
```suggestion
    const response = await fetch('/api/jf/watch.html/geolocation');
    const data: { country?: string } = await response.json();
    
    if (data.country) {
      const countryName = parseCountryName(data.country);
      setCountry(countryName);
      setCountryCode(data.country);
    }
```